### PR TITLE
rgw/admin: add support for new /info section

### PR DESCRIPTION
**Pull request by mattbenjamin**
Implement support for new /info AdminOps interface, which returns
an extensible json record similar to:

{"info":{"storage_backends":[{"name":"rados","cluster_id":"75d1938b-2949-4933-8386-fb2d1449ff03"}]}}

Signed-off-by: Matt Benjamin <mbenjamin@redhat.com>

## Checklist
- [ ] Added tests for features and functional changes
- [ ] Public functions and types are documented
- [ ] Standard formatting is applied to Go code
- [ ] Is this a new API? Is this new API marked PREVIEW?

### CODE REVIEW

**Comment by mattbenjamin** [comment]
@leseb I was having trouble running our own test suite locally, so decided to try with the go-ceph code

**Comment by phlogistonjohn** [comment]
Hi, thanks for the contribution! The CI is failing due to some of the doc comments not meeting the standard form. If you have any questions about the typical form for Go doc comment, please feel free to ask. 

We also have a new policy of tagging new API as preview. See the [stability doc](https://github.com/ceph/go-ceph/blob/master/docs/api-stability.md) and the [dev doc](https://github.com/ceph/go-ceph/blob/master/docs/development.md#documentation-conventions). Again, if you have any questions please just ask. Thanks!

File: rgw/admin/info.go [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+ * {"info":{"storage_backends":[{"name":"rados","cluster_id":"75d1938b-2949-4933-8386-fb2d1449ff03"}]}} */
+
+// Info struct
+type Backend struct {
+	Name    string `json:"name"`
+	ClusterID    string `json:"cluster_id"`
+}
+
+type Info struct {
+	Info map[string] []Backend
```
Comment (leseb) [inline rgw/admin/info.go:20]: I don't think this is a valid type. This works for me though:

```
	type StorageBackend struct {
		Name      string `json:"name"`
		ClusterID string `json:"cluster_id"`
	}

	type Info struct {
		InfoSpec struct {
			StorageBackends []StorageBackend `json:"storage_backends"`
		} `json:"info"`
	}

	info := []byte(`{"info":{"storage_backends":[{"name":"rados","cluster_id":"75d1938b-2949-4933-8386-fb2d1449ff03"}]}}`)

	u := Info{}
	err := json.Unmarshal(info, &u)
	assert.NoError(t, err)
	assert.Equal(t, "rados", u.InfoSpec.StorageBackends[0].Name)
```

**Review by leseb — CHANGES_REQUESTED** [review]

**Comment by leseb** [comment]
> Hi, thanks for the contribution! The CI is failing due to some of the doc comments not meeting the standard form. If you have any questions about the typical form for Go doc comment, please feel free to ask.
> 
> We also have a new policy of tagging new API as preview. See the [stability doc](https://github.com/ceph/go-ceph/blob/master/docs/api-stability.md) and the [dev doc](https://github.com/ceph/go-ceph/blob/master/docs/development.md#documentation-conventions). Again, if you have any questions please just ask. Thanks!

Do we really have to make all new API change preview?

**Comment by phlogistonjohn** [comment]
> Do we really have to make all new API change preview?

That's the current plan yes. There was a github [discussions thread](https://github.com/ceph/go-ceph/discussions/525) created a while back as well as some doc covering the recent changes: [policy](https://github.com/ceph/go-ceph/blob/master/docs/api-stability.md), [dev guide](https://github.com/ceph/go-ceph/blob/master/docs/development.md#documentation-conventions). This is the first cycle where we're "enforcing" the policy in the CI and code reviews.

If you have suggestions on changes to the policy I'd recommend you create a new thread in the discussion - I'm certainly open to changes but I want to make sure it's widely visible. But for the current PRs I'd like to be consistent and apply it evenly, I hope you understand. :-)

File: rgw/admin/info.go (hunk 1 above)
Comment (mattbenjamin) [reply rgw/admin/info.go:20]: go 1.14 compiled it, but I make no claims for it--just needed something to test with.
Do you want to take this over @leseb ?

**Comment by phlogistonjohn** [comment]
Hi all, it's been a while since this PR saw any activity. Based on previous comments I assume @mattbenjamin is not interested in driving this to completion. @leseb are you?  If not should/can we find someone who is?   It looks like a good thing to have and I think it would be good to get it in some form or another.

Thanks!

**Comment by mattbenjamin** [comment]
> Hi all, it's been a while since this PR saw any activity. Based on previous comments I assume @mattbenjamin is not interested in driving this to completion. @leseb are you? If not should/can we find someone who is? It looks like a good thing to have and I think it would be good to get it in some form or another.
> 
> Thanks!

oh, come on;  just busy

**Comment by leseb** [comment]
> Hi all, it's been a while since this PR saw any activity. Based on previous comments I assume @mattbenjamin is not interested in driving this to completion. @leseb are you? If not should/can we find someone who is? It looks like a good thing to have and I think it would be good to get it in some form or another.
> 
> Thanks!

I don't have time right now to take this over. Eventually, I might.

**Comment by mattbenjamin** [comment]
sorry, I'll try to hot this up

**Pull request closed without merging**